src/main/java/org/jabref/gui/maintable/MainTableHeaderRightClickMenu.java

```diff
@@ -0,0 +1,63 @@
+package org.jabref.gui.maintable;
+
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.RadioMenuItem;
+import javafx.scene.control.TableColumn;
+import javafx.scene.layout.StackPane;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.maintable.columns.MainTableColumn;
+import org.jabref.gui.preferences.table.TableTab;
+import org.jabref.gui.preferences.table.TableTabViewModel;
+import org.jabref.preferences.PreferencesService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainTableHeaderRightClickMenu extends ContextMenu {
+
+    public void show(MainTable mainTable) {
+        mainTable.setOnContextMenuRequested(clickEvent -> {
+
+            // Click on the tableColumns
+            if (!(clickEvent.getTarget() instanceof StackPane)) {
+
+                // Create radioMenuItemList from tableColumnList
+                List<RadioMenuItem> radioMenuItems = new ArrayList<>();
+                mainTable.getColumns().forEach(tableColumn -> radioMenuItems.add(createRadioMenuItem(tableColumn)));
+
+                // Clean items and add newItems
+                this.getItems().clear();
+
+                this.getItems().addAll(radioMenuItems);
+
+                // Show ContextMenu
+                this.show(mainTable, clickEvent.getScreenX(), clickEvent.getScreenY());
+            }
+
+
+            clickEvent.consume();
+        });
+    }
+
+    private RadioMenuItem createRadioMenuItem(TableColumn<BibEntryTableViewModel,?> tableColumn) {
+        RadioMenuItem radioMenuItem = new RadioMenuItem(((MainTableColumn<?>) tableColumn).getDisplayName());
+
+        radioMenuItem.setOnAction(event -> {
+
+            //return the column name when we click an item from context menu
+            System.out.println(((MainTableColumn) tableColumn).getModel().getName());
+        });
+
+        return radioMenuItem;
+    }
+
+    //Need help!!!
+    //We know the column name but cannot pass it in the parameters of the removeColumn()
+    public void removeColumns(MainTable mainTable, DialogService dialogService, PreferencesService preferencesService){
+        TableTab tableTab = new TableTab();
+        TableTabViewModel tableTabViewModel = new TableTabViewModel(dialogService, preferencesService);
+
+//        tableTabViewModel.removeColumn(tableTab.getList().getSelectionModel().getSelectedItem());
```

Comment:
We know how to access the column name but not sure if we can somehow use it in removeColumn( ).

Or is there a different way to remove a particular column from the preferences/entry table and maintable if we know the column name. 

Any suggestions would be highly appreciated!

Reply:
I would operate on the ColumnPreferences here as follows.
In MainTable you have the columns in this.columns and can map them to the MainTableColumnModel. And then you can create a new TablecolumnPreferences



```
   this.getColumns().stream()
                    .map(column -> (MainTableColumn<?>) column) 
```

e.g. (see PersistenceVisualStateTable)
```
preferences.storeMainTableColumnPreferences(new ColumnPreferences(
                mainTable.getColumns().stream()
                         .filter(col -> col instanceof MainTableColumn<?>)
                         .map(column -> ((MainTableColumn<?>) column).getModel())
                         .collect(Collectors.toList()),
                preferences.getColumnPreferences().getColumnSortOrder()));
```

Reply:
Thank you! We had a bit of confusion understanding how to create a new TablecolumnPreferences. We found another way to remove column though. Let us know if it works, thanks!